### ISSUE TITLE
Failed to install anchor-cli with cargo install --locked

### ISSUE THREAD

**Issue by piscespieces** [project-serum/anchor#984]
When trying to install the anchor-cli with the following command, it errors out at the end of the Building process:

- MacOS Big Sur (No M1)

```
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-m64" "-arch" "x86_64" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/cargo-install2k6Wlq/release/deps/anchor-9031490c7dbb6993.anchor.a2a48401-cgu.0.rcgu.o" "-L" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/cargo-install2k6Wlq/release/deps" "-L" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/cargo-install2k6Wlq/release/build/blake3-a325e896e0f9244f/out" "-L" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/cargo-install2k6Wlq/release/build/blake3-a325e896e0f9244f/out" "-L" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/cargo-install2k6Wlq/release/build/ring-661179a5bf967b5d/out" "-L" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/cargo-install2k6Wlq/release/build/zstd-sys-20daeaeea5d8b532/out" "-L" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/cargo-install2k6Wlq/release/build/hidapi-e666db647b354951/out" "-L" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/cargo-install2k6Wlq/release/build/bzip2-sys-0368f4dea75296aa/out/lib" "-L" "/Users/aeum3893/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/rustczxECdr/libbzip2_sys-6acb36e9b627991a.rlib" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/rustczxECdr/libzstd_sys-6d4a6be4aa0c9ec1.rlib" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/rustczxECdr/libring-6d97b6782501e94f.rlib" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/rustczxECdr/libhidapi-a3b3f025819fddd1.rlib" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/rustczxECdr/libblake3-2d4e4d380a6db8fd.rlib" "/Users/aeum3893/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-bc8e57a7c90d33c8.rlib" "-framework" "Security" "-framework" "Security" "-framework" "CoreFoundation" "-framework" "IOKit" "-framework" "CoreFoundation" "-framework" "AppKit" "-lc" "-liconv" "-lSystem" "-lresolv" "-lc" "-lm" "-liconv" "-L" "/Users/aeum3893/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "-o" "/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/cargo-install2k6Wlq/release/deps/anchor-9031490c7dbb6993" "-Wl,-dead_strip" "-nodefaultlibs"
note: ld: in /var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/rustczxECdr/libring-6d97b6782501e94f.rlib(chacha-x86_64-macosx.o), building for macOS, but linking in object file built for iOS Simulator,
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: failed to compile `anchor-cli v0.18.0 (https://github.com/project-serum/anchor#54f24e55)`, intermediate artifacts can be found at `/var/folders/v5/6yn_pvzx64s1rwh0zn55mp6r0000gp/T/cargo-install2k6Wlq`
```

No clue why this is happening, already tried reinstalling the `xcode-select` cli and ran `rustup update`

**Comment by piscespieces** [comment]
Tried with without the `--lock` flag, and got the following error output:

```
error[E0432]: unresolved import `clap::Clap`
  --> cli/src/lib.rs:10:5
   |
10 | use clap::Clap;
   |     ^^^^^^^^^^ no `Clap` in the root

error[E0432]: unresolved import `clap::Clap`
 --> cli/src/config.rs:4:5
  |
4 | use clap::Clap;
  |     ^^^^^^^^^^ no `Clap` in the root

error: cannot determine resolution for the derive macro `Clap`
  --> cli/src/config.rs:17:26
   |
17 | #[derive(Default, Debug, Clap)]
   |                          ^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot find attribute `clap` in this scope
  --> cli/src/config.rs:20:7
   |
20 |     #[clap(global = true, long = "provider.cluster")]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> cli/src/config.rs:23:7
   |
23 |     #[clap(global = true, long = "provider.wallet")]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot determine resolution for the derive macro `Clap`
  --> cli/src/lib.rs:48:17
   |
48 | #[derive(Debug, Clap)]
   |                 ^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot find attribute `clap` in this scope
  --> cli/src/lib.rs:49:3
   |
49 | #[clap(version = VERSION)]
   |   ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> cli/src/lib.rs:51:7
   |
51 |     #[clap(flatten)]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> cli/src/lib.rs:53:7
   |
53 |     #[clap(subcommand)]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot determine resolution for the derive macro `Clap`
  --> cli/src/lib.rs:57:17
   |
57 | #[derive(Debug, Clap)]
   |                 ^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot find attribute `clap` in this scope
  --> cli/src/lib.rs:62:11
   |
62 |         #[clap(short, long)]
   |           ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> cli/src/lib.rs:68:11
   |
68 |         #[clap(short, long)]
   |           ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> cli/src/lib.rs:71:11
   |
71 |         #[clap(short, long)]
   |           ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> cli/src/lib.rs:73:11
   |
73 |         #[clap(short, long)]
   |           ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> cli/src/lib.rs:77:11
   |
77 |         #[clap(short, long)]
   |           ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> cli/src/lib.rs:80:11
   |
80 |         #[clap(
   |           ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> cli/src/lib.rs:94:11
   |
94 |         #[clap(short, long)]
   |           ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> cli/src/lib.rs:98:11
   |
98 |         #[clap(short, long)]
   |           ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:101:11
    |
101 |         #[clap(
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:113:11
    |
113 |         #[clap(long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:117:11
    |
117 |         #[clap(long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:121:11
    |
121 |         #[clap(long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:125:11
    |
125 |         #[clap(long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:127:11
    |
127 |         #[clap(multiple_values = true)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:130:11
    |
130 |         #[clap(
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:142:11
    |
142 |         #[clap(subcommand)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:147:11
    |
147 |         #[clap(short, long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:157:11
    |
157 |         #[clap(short, long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:170:11
    |
170 |         #[clap(subcommand)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:191:11
    |
191 |         #[clap(
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:201:11
    |
201 |         #[clap(subcommand)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:208:11
    |
208 |         #[clap(long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:212:11
    |
212 |         #[clap(long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:215:11
    |
215 |         #[clap(
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot determine resolution for the derive macro `Clap`
   --> cli/src/lib.rs:225:17
    |
225 | #[derive(Debug, Clap)]
    |                 ^^^^
    |
    = note: import resolution is stuck, try simplifying macro imports

error: cannot determine resolution for the derive macro `Clap`
   --> cli/src/lib.rs:230:17
    |
230 | #[derive(Debug, Clap)]
    |                 ^^^^
    |
    = note: import resolution is stuck, try simplifying macro imports

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:235:11
    |
235 |         #[clap(short, long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:242:11
    |
242 |         #[clap(short, long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:249:11
    |
249 |         #[clap(short, long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:256:11
    |
256 |         #[clap(short, long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:265:11
    |
265 |         #[clap(short, long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:268:11
    |
268 |         #[clap(short, long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:275:11
    |
275 |         #[clap(short, long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:286:11
    |
286 |         #[clap(short, long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:289:11
    |
289 |         #[clap(short, long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:292:11
    |
292 |         #[clap(short = 't', long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> cli/src/lib.rs:300:11
    |
300 |         #[clap(short, long)]
    |           ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot determine resolution for the derive macro `Clap`
   --> cli/src/lib.rs:305:17
    |
305 | #[derive(Debug, Clap)]
    |                 ^^^^
    |
    = note: import resolution is stuck, try simplifying macro imports
```

Still looking for a fix

**Comment by piscespieces** [comment]
I got it to work, not sure why but I'll share what worked for me:

Opened a project in Xcode and followed these instructions

https://developer.apple.com/forums/thread/655316

From the Build Settings pane in XCode I changed the "Build Active Architecture Only" to YES. That worked for me. I was able to run the `cargo install --git https://github.com/project-serum/anchor --tag v0.18.0 anchor-cli --locked` command from my terminal and it did succeed.

**Comment by vinkim** [comment]
Just to add to this if someone else runs into the same problem.

I was running MacOS Big Sur aswell, and not the newest Xcode. After installing MacOS Monterey (Version 12.2) and latest available Xcode - Version 13.1(13A1030d), it successfully was able to compile with the
`cargo install --git https://github.com/project-serum/anchor --tag v0.18.0 anchor-cli --locked` command.
